src/main/java/telraam/logic/lapper/slapper/Slapper.java

```diff
@@ -0,0 +1,139 @@
+package telraam.logic.lapper.slapper;
+
+import io.dropwizard.jersey.setup.JerseyEnvironment;
+import org.jdbi.v3.core.Jdbi;
+import telraam.database.daos.LapSourceDAO;
+import telraam.database.models.Detection;
+import telraam.database.models.LapSource;
+import telraam.logic.lapper.Lapper;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+// Lapper that only uses a single sql query
+public class Slapper implements Lapper {
+    private final String SOURCE_NAME = "slapper";
+    private final int DEBOUNCE_TIMEOUT = 10;
+    private final ScheduledExecutorService scheduler;
+    private boolean debounceScheduled;
+    private final Logger logger;
+    private final Jdbi jdbi;
+
+    public Slapper(Jdbi jdbi) {
+        this.jdbi = jdbi;
+        this.scheduler = Executors.newScheduledThreadPool(1);
+        this.logger = Logger.getLogger(Slapper.class.getName());
+        this.debounceScheduled = false;
+
+        // Get the lapSourceId, create the source if needed
+        LapSourceDAO lapSourceDAO = jdbi.onDemand(LapSourceDAO.class);
+        if (lapSourceDAO.getByName(SOURCE_NAME).isEmpty()) {
+            lapSourceDAO.insert(new LapSource(SOURCE_NAME));
+        }
+    }
+
+    @Override
+    public void handle(Detection msg) {
+        if (!this.debounceScheduled) {
+            this.debounceScheduled = true;
+            this.scheduler.schedule(() -> {
+                try {
+                    this.calculateLaps();
+                } catch (Exception e) {
+                    logger.severe(e.getMessage());
+                }
+                this.debounceScheduled = false;
+            }, DEBOUNCE_TIMEOUT, TimeUnit.SECONDS);
+        }
+    }
+
+    private void calculateLaps() {
+        logger.info("Slapper: Calculating laps...");
+        this.jdbi.useHandle(handle -> handle.execute(
+                        """
+                                WITH switchovers AS (
+                                	SELECT teamid AS team_id,
+                                		newbatonid,
+                                		timestamp AS start_timestamp,
+                                		COALESCE(
+                                			LEAD(timestamp) OVER (PARTITION BY teamid ORDER BY timestamp),
+                                			timestamp + INTERVAL '1 MONTH'
+                                		) AS next_baton_switch
+                                	FROM batonswitchover
+                                ),
+                                team_detections AS (
+                                	SELECT MIN(station_id) AS station_id,
```

Comment:
Does this assume that station_id's are ordered in the order they are on the parcours (instead of using the distance metric?

Reply:
I see two ways to address this:
- (correct way, imo): make each lapper use the distance field instead of hardcoding station IDs. That way, we'll also be flexible to the amount of stations
- Alternatively, assert that the order of station_id's is the same as the distance order, somewhere in the Telraam codebase, as we've already been bit by this once (RobustLapper does this the 'correct' way, but we once forgot to fill in the distance fields with realistic data)

Reply:
I've removed the `MIN(station_id)` as it didn't have any effect.
[ae537f8](https://github.com/12urenloop/Telraam/pull/133/commits/ae537f80530c923f8e6932e996e4ff64e6b1172e).
When it comes to making other lappers use the distance field, I suggest making an issue ;)